pages/documentation/how-pyth-works/weekly-schedule-format.mdx

```diff
@@ -0,0 +1,37 @@
+# Pyth Oracle Weekly Schedule Format, version 1
+
+This document outlines the rules for specifying contents of a new Pyth product metadata field - `weekly_schedule` . The field specifies the recurring weekly schedule of a product’s market hours. It serves as a reference for `pyth-agent` to stop publishing outside the hours specified in the schedule. Notable use cases include:
+
+- FX
+- Metals
+- Stocks
+
+## The format
+
+```plain
+Timezone,MHKind,MHKind,MHKind,MHKind,MHKind,MHKind,MHKind
+```
+
+**Note: None of the comma-separated values can be ommitted - exactly one timezone and seven subsequent per-day schedules are required. That said, ommitting `weekly_schedule` on a symbol is allowed and will cause pyth-agent to default to 24/7 publishing (the usual behavior before this feature)**
+
+**Detailed Definitions**
+
+- `Timezone` - A human-readable tz database TZ identifier of the market’s local timezone - e.g. `America/New_York`. **Full list of identifiers can be found [here](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)**. Notes:
+  - Daylight-saving time - Handled automatically by pyth-agent code dealing with the format.
+- `MHKind` - A single week day’s schedule. The `MHKind` values describe week days from Monday to Sunday, in that order. `MHKind` is defined as one of:
+  - `O` - all-day open
+  - `C` - all-day closed
+  - `Hour:Minute-Hour:Minute` - specific open and close times in the market-local timezone. Open time must come before close time. **The range is inclusive.** Definitions:
+    - `Hour` - number of hours from `00` to `24` . Notes:
+      - Leading zeros are optional - e.g. `9` and `09`, `0` and `00` are equivalent.
+      - `24` can only be used to specify `24:00`. This value is used to express the final moment of a given day (split-second before `00:00` on the next day). Context: Without this special case, the next best thing would be `23:59` which could cause a symbol to go down between `23:59` and the next day’s `00:00` for a full minute.
+    - `Minute` - number of minutes from `00` to `59`. Notes:
+      - Leading zeros are **mandatory** - e.g. `9:05`, `9:00`, `15:07`
```

Comment:
> - Leading zeros are **mandatory**

I understood this as `09:05` maybe just me but could be confusing?

Reply:
yeah took me a sec to understand why some leading zeros were mandatory while other optional
guess it refers to hours vs minutes but poor structure in the review part here

imo thats fine when it will be on a real website